src/main/java/commands/Command.java

```diff
@@ -0,0 +1,22 @@
+package commands;
+import java.util.function.Function;
+
+import client.Rock;
+import io.Parser;
+/**
+ * Representation of a command
+ * that can be issued to the chatbot.
+ * @author Alvis Ng (supermii2)
+ */
+public abstract class Command implements Function<Parser, String> {
```

Comment:
Is there a reason for the class Command to implement Function? Since it seems like the default apply(object) method is never used, correct me if I'm wrong.

Reply:
Each command is expected to return a response to the user (a response) to let the user know the task has been completed, which is why it implements a function thats takes in a parsed input and outputs a string. The apply method is used in Invoker::handle